### PR TITLE
Sync Committee RPC Changes to Validator Service for Altair

### PR DESCRIPTION
**Pull request by rauljordan**
Part of #8638, this PR adds the changes to beacon-chain/rpc/prysm/v1alpha1/validator sync committee and proposer sync committee changes from the hf1 branch into develop. This includes changes for validators to:

- Submit sync committee contributions
- Request sync committee information
- Deduplication algorithm for proposer sync contribution information

### CODE REVIEW

File: beacon-chain/rpc/prysm/v1alpha1/validator/proposer_sync_aggregate.go [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+}
+
+// mostProfitable returns the most profitable sync contribution, the one with the most
+// votes (ie. aggregation bits count)
+func (cs proposerSyncContributions) mostProfitable() *eth.SyncCommitteeContribution {
+	if len(cs) == 0 {
+		return nil
+	}
+	mostProfitable := cs[0]
+	for _, c := range cs[1:] {
```
Comment (rauljordan) [inline beacon-chain/rpc/prysm/v1alpha1/validator/proposer_sync_aggregate.go:85]: length check?

File: beacon-chain/rpc/prysm/v1alpha1/validator/sync_committee.go [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+		Root: r,
+	}, nil
+}
+
+// SubmitSyncMessage submits the sync committee message to the network.
+// It also saves the sync committee message into the pending pool for block inclusion.
+func (vs *Server) SubmitSyncMessage(ctx context.Context, msg *ethpb.SyncCommitteeMessage) (*emptypb.Empty, error) {
+	errs, ctx := errgroup.WithContext(ctx)
+
+	idxResp, err := vs.syncSubcommitteeIndex(ctx, msg.ValidatorIndex, msg.Slot)
```
Comment (rauljordan) [inline beacon-chain/rpc/prysm/v1alpha1/validator/sync_committee.go:39]: Would be good to look at how many calls this triggers and perhaps could use a cache here

**Review by rauljordan — COMMENTED** [review]
Blocked by #9365

File: beacon-chain/rpc/prysm/v1alpha1/validator/sync_committee.go (hunk 2 above)
Comment (terencechain) [reply beacon-chain/rpc/prysm/v1alpha1/validator/sync_committee.go:39]: This is pretty well cached underneath. In the current spec there's only 512 sync committee participants active at a given time. Chances are low to pursue further overish optimization

**Comment by rauljordan** [comment]
Closing as already done in smaller pieces

**Pull request closed without merging**